Handle broken symlinks

danrue opened the pull request:
When phockup.py finds a broken symlink and tries to shutil.copy2() it, a
traceback such as the following is produced:

    eagle/Pictures/iChat IconsFile not found: eagle/Pictures/iChat Icons
    Traceback (most recent call last):
      File "/home/drue/bin/phockup", line 75, in <module>
        main(sys.argv[1:])
      File "/home/drue/bin/phockup", line 69, in main
        date_regex=date_regex
      File "/home/drue/phockup/src/phockup.py", line 34, in __init__
        self.walk_directory()
      File "/home/drue/phockup/src/phockup.py", line 65, in walk_directory
        self.process_file(file)
      File "/home/drue/phockup/src/phockup.py", line 153, in process_file
        shutil.copy2(file, target_file)
      File "/usr/lib/python3.6/shutil.py", line 257, in copy2
        copyfile(src, dst, follow_symlinks=follow_symlinks)
      File "/usr/lib/python3.6/shutil.py", line 120, in copyfile
        with open(src, 'rb') as fsrc:
    FileNotFoundError: [Errno 2] No such file or directory: 'eagle/Pictures/iChat Icons'

Instead of failing, phockup should skip over the file:

    eagle/Pictures/iChat Icons => skipped, no such file or directory

ivandokov commented:
Thanks for your contribution!
Can you please add the same check to the `move` method and probably add a test for this?

danrue commented:
Good idea @ivandokov. Added the exception handler to move() and added a few tests.

BTW thanks for this utility. It was exactly what I was looking for!
